R/jitter.R

```diff
@@ -190,3 +203,77 @@ jitter <- function(dir = getwd(),
   }
   return(invisible(likesaved))
 }
+
+#' Accomplish a single jittered model run
+#'
+#' @param i Index of the jitter. This does not strictly have to be an integer.
+#' @param dir Directory of the base model to be jittered
+#' @param printlikes 
+#' @param jitter_fraction 
+#' @param init_values_src 
+#' @param exe 
+#' @param verbose 
+#' @param ... Additional arguments passed to [r4ss::run()]
+#'
+#' @return
+#' @export
+#'
+#' @examples
+iterate_jitter <- function(i, 
+                           dir = getwd(),
+                           printlikes = TRUE,
+                           jitter_fraction = NULL,
+                           init_values_src = NULL,
+                           exe = "ss",
+                           verbose = FALSE,
+                           ...) {
+  jitter.dir <- file.path(dir, paste0('jitter', i))
+  copy_SS_inputs(dir.old = dir, dir.new = jitter.dir, overwrite = TRUE)
+  # check for use of .par file and replace original if needed
+  if (starter[["init_values_src"]] == 1) {
+    file.copy(from = "ss.par_0.sso", to = file.path(jitter.dir, "ss.par"), 
+              overwrite = TRUE)
+  }
+  # run model
+  r4ss::run(dir = jitter.dir, exe = exe, verbose = verbose, ...)
+  # Only save stuff if it converged
+  if ("Report.sso" %in% list.files(path = jitter.dir)) {
+    rep <- SS_read_summary(file.path(jitter.dir, 'ss_summary.sso'))
+    if (is.null(rep)) {
+      report <- SS_output(
+        dir = jitter.dir, forecast = FALSE,
+        covar = FALSE, NoCompOK = TRUE,
+        verbose = verbose, warn = verbose, hidewarn = !verbose, printstats = verbose
+      )
+      like <- report[["likelihoods_used"]][row.names(report[["likelihoods_used"]]) == "TOTAL", "values"]
+    } else {
+      like <- rep[["likelihoods"]][grep("TOTAL", row.names(rep[["likelihoods"]])), 1]
+    }
+    if (printlikes) {
+      message("Likelihood for jitter ", i, " = ", like)
+    }
+    # move output files to base model directory
+    sso.files <- grep("^[CcPRw][a-zA-Z]+\\.sso|summary\\.sso|\\.par$", 
+                      list.files(jitter.dir), value = TRUE)
+    sapply(sso.files, \(x) 
+           file.copy(from = file.path(jitter.dir, x), 
+                     to = file.path(dir, gsub('\\.', paste0(i, '\\.'), x)),
+                     overwrite = TRUE))
+    # delete jitter model directory
+    unlink(jitter.dir, recursive = TRUE)
+    return(like)
+  } else {
+    unlink(jitter.dir, recursive = TRUE)
+    if (verbose) warning("No Report.sso file found from run ", i)
+  }
+}
+
+plan(sequential)
```

Comment:
could you move these lines to the `@examples` section at the top? Currently it tries to run these when loading the package.